components/binance/browser/binance_protocol_handler.cc

```diff
@@ -6,47 +6,28 @@
 #include "brave/components/binance/browser/binance_protocol_handler.h"
 
 #include <string>
+#include <map>
 #include <utility>
 
 #include "base/strings/strcat.h"
 #include "base/strings/string_util.h"
 #include "base/task/post_task.h"
 #include "brave/common/url_constants.h"
+#include "brave/components/binance/browser/binance_service.h"
+#include "components/user_prefs/user_prefs.h"
 #include "content/public/browser/browser_task_traits.h"
 #include "content/public/browser/browser_thread.h"
 #include "net/base/escape.h"
+#include "net/base/url_util.h"
 
 namespace {
 
-GURL TranslateUrl(const GURL& url) {
-  if (!url.is_valid()) {
-    return GURL();
-  }
-
-  std::string path = url.path();
-  std::string query;
-
-  if (url.has_query()) {
-    query = base::StrCat({
-      "?",
-      net::EscapeExternalHandlerValue(url.query())
-    });
-  }
-
-  base::ReplaceFirstSubstringAfterOffset(&path, 0, "/", "");
-  return GURL(
-      base::StrCat({
-        "chrome://newtab",
-        path,
-        query
-      }));
-}
-
 void LoadNewTabURL(
     const GURL& url,
     content::WebContents::OnceGetter web_contents_getter,
     ui::PageTransition page_transition,
-    bool has_user_gesture) {
+    bool has_user_gesture,
+    const base::Optional<url::Origin> initiating_origin) {
```

Comment:
Should this be a const reference? I saw that we're doing that elsewhere in here.

Reply:
I'm not 100% sure, I removed it because I wasn't sure if it could cause problems with the bind, but I know the url one does it already. So I'll update to it.